src/nvim/os/fs.c

```diff
@@ -391,6 +391,37 @@ int os_open(const char* path, int flags, int mode)
   return r;
 }
 
+/// Set a file to close on exec.
+//
+// @return -1 if failed to set, 0 otherwise.
+int os_set_cloexec(const int fd)
+{
+#ifdef HAVE_FD_CLOEXEC
+  int e;
+  int fdflags = fcntl(fd, F_GETFD);
+  if (fdflags < 0) {
+    e = errno;
+    ELOG("Failed to get flags on descriptor %d: %s", fd, strerror(errno));
```

Comment:
> Were you originally saying that hypothetically, if the ELOG macro could set errno, then we would have to use e in the call to strerror(). Then in the second comment saying that since it doesn't, we're fine as we are.

Yes, my concern was about the other parameters to `do_log()` that could change `errno`.

> While checking just now if it's possible that strerror() changes errno, I noticed that in posix 2004 the return value of strerror() when an error occurs isn't specified, so it could be NULL. I'll add a check for that.

Linux man page says C99 and POSIX.1-2008 require the return value of `strerror()` to be non-NULL.